auth-jwt-service/src/main/java/io/stargate/auth/jwt/JwtValidator.java

```diff
@@ -0,0 +1,78 @@
+package io.stargate.auth.jwt;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.jwk.source.JWKSource;
+import com.nimbusds.jose.jwk.source.RemoteJWKSet;
+import com.nimbusds.jose.proc.BadJOSEException;
+import com.nimbusds.jose.proc.JWSKeySelector;
+import com.nimbusds.jose.proc.JWSVerificationKeySelector;
+import com.nimbusds.jose.proc.SecurityContext;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.proc.ConfigurableJWTProcessor;
+import com.nimbusds.jwt.proc.DefaultJWTProcessor;
+import io.stargate.auth.UnauthorizedException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.text.ParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JwtValidator {
+
+  private static final Logger logger = LoggerFactory.getLogger(JwtValidator.class);
+
+  // Set up a JWT processor to parse the tokens and then check their signature
+  // and validity time window (bounded by the "iat", "nbf" and "exp" claims)
+  private final ConfigurableJWTProcessor<SecurityContext> jwtProcessor;
+
+  public JwtValidator(String providerURL) throws MalformedURLException {
+    this.jwtProcessor = new DefaultJWTProcessor<>();
+
+    // Pull the public RSA keys from the provided well-known URL to validate the JWT signature.
+    JWKSource<SecurityContext> keySource =
+        new RemoteJWKSet<>(
+            new URL(providerURL)); // by default this will cache the JWK for 15 minutes
```

Comment:
is it configured somewhere? Is there a way to make it explicit?

Reply:
I can but feels excessive to pass in already default config.